### PR TITLE
bunch of small language fixes

### LINKED ISSUE #248: lang/c files needs to be updated to use the new code_functions and code_libraries lists
see comments in https://github.com/knausj85/knausj_talon/pull/240, specifically https://github.com/knausj85/knausj_talon/pull/240#discussion_r493987137

> This was probably supposed to be `c_functions`.
> 
> This is probably be out of scope for this, but I think this should probably be moved over to use the `code_functions` and `code_libraries` lists, as done for python and R. And maybe move to the new code_insert_function stuff, which I think can basically be copied from csharp while we're here, or open an item for it. This way, we benefit from the imgui and stuff in code.py.
> 
> https://github.com/knausj85/knausj_talon/blob/master/lang/csharp/csharp.py

### PR DESCRIPTION
**Pull request by fidgetingbits**
this includes a fix #248, a couple automatic white space changes, a typo in csharp, some additional python stuff for adding docstring documentation as well as type hinting. 

also changed the way the generic library command worked in programmin.talon is because it seems specific to r? now I make it possible in programming.talon to just to add from a list of available includes or libraries or whatever, which makes it a little bit harder to generically do because the terminology differs for different programing languages, so it may be something we don't want generic?

### CODE REVIEW

File: lang/python/python.py [hunk 1]
```diff
@@ -23,6 +22,48 @@
     "update": "update",
 }
 
+"""a set of fields used in python dog strings that will follow the
```
Comment (dwiel) [inline lang/python/python.py:25]: Dog strings

File: lang/programming.talon [hunk 2]
```diff
@@ -44,9 +44,7 @@ toggle funk: user.code_toggle_functions()
 toggle library: user.code_toggle_libraries()
 funk <user.code_functions>:
     user.code_insert_function(code_functions, "")
-library <user.code_libraries>:
-    insert("library()")
-    key(left)
+(include|import|library) <user.code_libraries>:
```
Comment (dwiel) [inline lang/programming.talon:47]: I agree, seems like this should be language specific. Include doesnt make sense in python, import doesnt in c, etc.

File: lang/python/python.talon [hunk 3]
```diff
@@ -93,16 +93,32 @@ action(user.code_true): "True"
...
+    insert("{python_type_list}")
+dock {user.python_docstring_fields}:
+    insert("{python_docstring_fields}")
+    edit.left()
+dock type {user.python_type_list}:
+    user.insert_cursor(":type [|]: {python_type_list}")
+dock returns type {user.python_type_list}:
+    user.insert_cursor(":rtype [|]: {python_type_list}")
+
+dunder in it: "__init__"
```
Comment (dwiel) [inline lang/python/python.talon:121]: Why move these, just makes mergers harder

File: lang/programming.talon (hunk 2 above)
Comment (fidgetingbits) [reply lang/programming.talon]: i moved it to per file

File: lang/python/python.talon (hunk 3 above)
Comment (fidgetingbits) [reply lang/python/python.talon:120]: good call. was just effect of my python file being way more full of cruft and and trying to keep in the original stuff, but I think it should be fix now.

Comment (dwiel) [reply lang/python/python.talon:120]: Cool thanks

File: lang/c/c.py [hunk 4]
```diff
@@ -227,25 +210,60 @@ def stdint_signed(m) -> str:
...
+            )
+        )
+
+        actions.user.code_insert_function(result, None)
+
+    def code_insert_library(text: str, selection: str):
+        actions.insert("#include <>")
+        actions.edit.left()
+        actions.clip.set_text(text + "{}".format(selection))
+        actions.edit.paste()
```
Comment (knausj85) [inline lang/c/c.py:269]: I'd suggest using user.paste, which preserves the clipboard.

```
actions.user.paste("include <{}>".format(selection)
```

**Pull request merged**